camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -222,8 +226,24 @@ public void run() throws Exception
       {
         System.err.println("unable to process " + topicName + " skipping...");
         e.printStackTrace();
-        executorService.shutdown();
-        throw e;
+      }
+    }
+
+    Future<?> completedTask;
+    int remainingTaskCount = tasksToComplete.size();
+
+    while (remainingTaskCount > 0) {
+      completedTask = completionService.take();
+      remainingTaskCount--;
+
+      try {
+          completedTask.get();
+      } catch (ExecutionException e) {
+          // If a task completed with an exception shutdown all other tasks and exit
+          // re-raise the exception to make the parent job fail.
+          executorService.shutdown();
```

Comment:
> However a few lines down we'll be waiting for their termination

but line below throws an Exception, so `awaitTermination` won't be reached. not sure if this is bad, maybe not much.